--- a/tachidesk-server-electron-launcher.sh
+++ b/tachidesk-server-electron-launcher.sh
@@ -1,3 +1,3 @@
 #!/bin/sh
 
-exec /usr/bin/java -Dsuwayomi.tachidesk.config.server.webUIInterface=electron -Dsuwayomi.tachidesk.config.server.electronPath=/usr/bin/electron -jar /usr/share/java/tachidesk-server/tachidesk-server.jar
+exec /usr/bin/java -Dsuwayomi.tachidesk.config.server.webUIInterface=electron -Dsuwayomi.tachidesk.config.server.electronPath=/usr/share/electron/electron -jar /usr/share/java/tachidesk-server/tachidesk-server.jar
